lib/swagger.js

```diff
@@ -21,37 +21,64 @@ exports.run = function({ args, opts }) {
     return Promise.reject(new Error('No api key provided. Please use --key'));
   }
 
-  if (!args[0]) {
-    return Promise.reject(
-      new Error('No swagger file provided. Usage `rdme swagger <swagger-file>`'),
-    );
-  }
+  function callApi(specPath) {
+    function success(data) {
+      console.log(data);
+      console.log('Success! '.green);
+    }
 
-  function success(data) {
-    console.log(data);
-    console.log('Success! '.green);
-  }
+    function error(err) {
+      try {
+        if (err.statusCode === 401) {
+          return Promise.reject(
+            new Error('There was a problem with your api key. Please try again.'),
+          );
+        }
 
-  function error(err) {
-    try {
-      return Promise.reject(new Error(JSON.parse(err.error).description));
-    } catch (e) {
-      return Promise.reject(new Error('There was an error uploading!'));
+        return Promise.reject(new Error(JSON.parse(err.error).description));
+      } catch (e) {
+        return Promise.reject(new Error('There was an error uploading!'));
+      }
     }
-  }
 
-  const options = {
-    formData: {
-      swagger: fs.createReadStream(path.resolve(process.cwd(), args[0])),
-    },
-    auth: { user: key },
-  };
+    const options = {
+      formData: {
+        swagger: fs.createReadStream(path.resolve(process.cwd(), specPath)),
+      },
+      auth: { user: key },
+    };
+
+    // Create
+    if (!id) {
+      return request.post(`${config.host}/api/v1/swagger`, options).then(success, error);
+    }
 
-  // Create
-  if (!id) {
-    return request.post(`${config.host}/api/v1/swagger`, options).then(success, error);
+    // Update
+    return request.put(`${config.host}/api/v1/swagger/${id}`, options).then(success, error);
   }
 
-  // Update
-  return request.put(`${config.host}/api/v1/swagger/${id}`, options).then(success, error);
+  if (args[0]) {
+    return callApi(args[0]);
+  }
+
+  // If the user didn't supply a specification, let's try to locate what they've got, and upload
+  // that. If they don't have any, let's let the user know how they can get one going.
+  return new Promise((resolve, reject) => {
+    ['swagger.json', 'swagger.yaml', 'openapi.json', 'openapi.yaml'].forEach(function(file) {
+      if (!fs.existsSync(file)) {
+        return;
+      }
+
+      console.log(`We found ${file} and are attempting to upload it.`.yellow);
+
+      resolve(callApi(file));
+    });
+
+    reject(
+      new Error(
```

Comment:
I feel like if people are running this command, they likely have a swagger file, maybe it would be better to tell them how to provide a path as well? Something like “We couldn’t find a swagger or OpenAPI file. Run `rdme swagger ./path/to/file` to upload an existing file or `rdme oas init` to create a fresh one!”

Reply:
Good idea.